src/pkg/runner/runner.go

```diff
@@ -175,6 +180,63 @@ func (r *Runner) importTasks(includes []map[string]string, dir string, setVariab
 	return nil
 }
 
+func (r *Runner) loadIncludedTaskFile(taskName string) (string, error) {
+	// Check if running task directly from included task file
+	includedTask := strings.Split(taskName, ":")
+	if len(includedTask) == 2 {
```

Comment:
Lots of nested logic/loops going on in this block, can we break this out into functions?

Reply:
broken up into functions where it made sense. updated in latest commit